src/pip/_internal/resolution/resolvelib/found_candidates.py

```diff
@@ -111,7 +127,17 @@ def __init__(
         self._prefers_installed = prefers_installed
         self._incompatible_ids = incompatible_ids
 
+    @overload
     def __getitem__(self, index: int) -> Candidate:
+        ...
+
+    @overload
+    def __getitem__(self, index: slice) -> SequenceCandidate:
+        ...
+
+    def __getitem__(
+        self, index: Union[int, slice]
+    ) -> Union[Candidate, SequenceCandidate]:
```

Comment:
Is the overload really needed? Since this is not going to be called anywhere anyway, I feel we can be lazy and just do `__getitem__(self, index: Any) -> Any` (or use `Union`; that’s not too complex so why not).